tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -354,8 +363,29 @@ func testAnsiblePing(t *testing.T) {
 
 // testAzureFirewallRule asserts if the created instance has firewall rule that opens container logs port.
 func testAzureInstancesFirewallRule(t *testing.T) {
+	var err error
+	windowsVM, err = getAzureCreds()
+	require.NoErrorf(t, err, "failed to obtain credentials for %s", credentials.GetInstanceId())
+	testInstanceFirewallRule(t)
+}
+
+func getAzureCreds() (windowsVM types.WindowsVM, err error) {
+	w := &types.Windows{}
+	ipAddress, password, err := getIpPass(credentials.GetInstanceId())
+	credentials = types.NewCredentials(credentials.GetInstanceId(), ipAddress, password, azureUser)
+	w.Credentials = credentials
+	return w, err
+}
+
+func testAzureSSHConnection(t *testing.T) {
 	ipAddress, password, err := getIpPass(credentials.GetInstanceId())
 	require.NoErrorf(t, err, "failed to obtain credentials for %s", credentials.GetInstanceId())
 	credentials = types.NewCredentials(credentials.GetInstanceId(), ipAddress, password, azureUser)
-	testInstanceFirewallRule(t)
+	config := &ssh.ClientConfig{
+		User:            credentials.GetUserName(),
+		Auth:            []ssh.AuthMethod{ssh.Password(credentials.GetPassword())},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	_, err = ssh.Dial("tcp", credentials.GetIPAddress()+":22", config)
```

Comment:
Can you check if the object if not nil or empty? Perhaps there is nothing wrong in doing a windowsVM.Run(some comman) to prove it out?